### PR TITLE
Feat/suffix separator variable

### PR DESCRIPTION
**Pull request by thiagonache**
Add new variable to handle hostname separator char in the `compute_instance` and `umig` modules.
Discussed on issue https://github.com/terraform-google-modules/terraform-google-vm/issues/217

### CODE REVIEW

**Comment by thiagonache** [comment]
I've signed the CLA few seconds ago.

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_generate_modules
 The modules need to be regenerated. Please run `make_build`. 
 ```bash 
Checking submodule's files generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/compute_instance/README.md /tmp/tmp.7pyw14S2p5/workspace/modules/compute_instance/README.md
20a21
> | hostname\_suffix\_separator | Separator charactere to compose hostname when add\_hostname\_suffix is set to true. | `string` | `"-"` | no |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/umig/README.md /tmp/tmp.7pyw14S2p5/workspace/modules/umig/README.md
20a21
> | hostname\_suffix\_separator | Separator charactere to compose hostname when add\_hostname\_suffix is set to true. | `string` | `"-"` | no |
Error: submodule's files generation has not been run, please run the
'make build' command and commit changes
 ``` 
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/compute_instance/README.md /tmp/tmp.Tj6TsrcJfS/generate_docs/workspace/modules/compute_instance/README.md
20a21
> | hostname\_suffix\_separator | Separator charactere to compose hostname when add\_hostname\_suffix is set to true. | `string` | `"-"` | no |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/umig/README.md /tmp/tmp.Tj6TsrcJfS/generate_docs/workspace/modules/umig/README.md
20a21
> | hostname\_suffix\_separator | Separator charactere to compose hostname when add\_hostname\_suffix is set to true. | `string` | `"-"` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ```

**Comment by thiagonache** [comment]
I believe the error is not related to my code change. Can you check it, please @morgante ?
```
-----> Converging <disk-snapshot-local>...
$$$$$$ Reading the Terraform client version...
       Terraform v1.0.5
       on linux_amd64
       + provider registry.terraform.io/hashicorp/google v4.4.0
       + provider registry.terraform.io/hashicorp/null v3.1.0
       + provider registry.terraform.io/hashicorp/random v2.3.1
       
       Your version of Terraform is out of date! The latest version
       is 1.1.1. You can update by downloading from https://www.terraform.io/downloads.html
$$$$$$ Finished reading the Terraform client version.
$$$$$$ Verifying the Terraform client version is in the supported interval of >= 0.11.4, < 1.1.0...
$$$$$$ Finished verifying the Terraform client version.
$$$$$$ Selecting the kitchen-terraform-disk-snapshot-local Terraform workspace...
$$$$$$ Finished selecting the kitchen-terraform-disk-snapshot-local Terraform workspace.
$$$$$$ Downloading the modules needed for the Terraform configuration...
       - disk_snapshot in ../../../../examples/compute_instance/disk_snapshot
       - disk_snapshot.compute_instance in ../../../../modules/compute_instance
       - disk_snapshot.disk_snapshots in ../../../../modules/compute_disk_snapshot
       - disk_snapshot.instance_template in ../../../../modules/instance_template
$$$$$$ Finished downloading the modules needed for the Terraform configuration.
$$$$$$ Validating the Terraform configuration files...
       ╷
       │ Warning: Version constraints inside provider configuration blocks are deprecated
       │ 
       │   on network.tf line 18, in provider "random":
       │   18:   version = "~> 2.2"
       │ 
       │ Terraform 0.13 and earlier allowed provider version constraints inside the
       │ provider configuration block, but that is now deprecated and will be
       │ removed in a future version of Terraform. To silence this warning, move the
       │ provider version constraint into the required_providers block.
       ╵
       Success! The configuration is valid, but there were some
       validation warnings as shown above.
       
$$$$$$ Finished validating the Terraform configuration files.
$$$$$$ Building the infrastructure based on the Terraform configuration...
       ╷
       │ Warning: Version constraints inside provider configuration blocks are deprecated
       │ 
       │   on network.tf line 18, in provider "random":
       │   18:   version = "~> 2.2"
       │ 
       │ Terraform 0.13 and earlier allowed provider version constraints inside the
       │ provider configuration block, but that is now deprecated and will be
       │ removed in a future version of Terraform. To silence this warning, move the
       │ provider version constraint into the required_providers block.
       ╵
       ╷
       │ Error: Invalid function argument
       │ 
       │   on ../../../../modules/instance_template/main.tf line 97, in resource "google_compute_instance_template" "tpl":
       │   97:       email  = lookup(service_account.value, "email", null)
       │     ├────────────────
       │     │ service_account.value is null
       │ 
       │ Invalid value for "inputMap" parameter: argument must not be null.
       ╵
       ╷
       │ Error: Invalid function argument
       │ 
       │   on ../../../../modules/instance_template/main.tf line 98, in resource "google_compute_instance_template" "tpl":
       │   98:       scopes = lookup(service_account.value, "scopes", null)
       │     ├────────────────
       │     │ service_account.value is null
       │ 
       │ Invalid value for "inputMap" parameter: argument must not be null.
```

**Comment by thiagonache** [comment]
I just tested out the master branch and it does fail with the same errors.

**Comment by thiagonache** [comment]
@morgante I started to troubleshoot this and I've found that the example passes the service account as `null` (it was added 3 months ago)
```
module "instance_template" {
  source          = "../../../modules/instance_template"
  region          = var.region
  project_id      = var.project_id
  subnetwork      = var.subnetwork
  name_prefix     = "instance-disk-snapshot"
  service_account = null
```

but the module fails on the dynamic block that was added three years ago.
```
  dynamic "service_account" {
    for_each = [var.service_account]
    content {
      email  = lookup(service_account.value, "email", null)
      scopes = lookup(service_account.value, "scopes", null)
    }
  }
```

I don't know how to fix it properly. I've added `service_account = { email = "", scopes = ["cloud-platform"] }`. Please, let me know.

**Comment by thiagonache** [comment]
@bharathkkb @morgante I don't know how it was working before but everything seems okay now. Can you help me to get this code merged, please? 😊

File: modules/umig/README.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ See the [simple](https://github.com/terraform-google-modules/terraform-google-vm
 | access\_config | Access configurations, i.e. IPs via which the VM instance can be accessed via the Internet. | <pre>list(list(object({<br>    nat_ip       = string<br>    network_tier = string<br>  })))</pre> | `[]` | no |
 | additional\_networks | Additional network interface details for GCE, if any. | <pre>list(object({<br>    network            = string<br>    subnetwork         = string<br>    subnetwork_project = string<br>    network_ip         = string<br>    access_config = list(object({<br>      nat_ip       = string<br>      network_tier = string<br>    }))<br>  }))</pre> | `[]` | no |
 | hostname | Hostname of instances | `string` | `""` | no |
+| hostname\_suffix\_separator | Separator charactere to compose hostname when add\_hostname\_suffix is set to true. | `string` | `"-"` | no |
```
Comment (bharathkkb) [inline modules/umig/README.md:21]:
```suggestion
| hostname\_suffix\_separator | Separator character to compose hostname when add\_hostname\_suffix is set to true. | `string` | `"-"` | no |
```

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @thiagonache 
I believe we don't test `disk-snapshot-local` in CI and hence this was not caught. CI was failing due to another flaky issue which I have opened a fix for in https://github.com/terraform-google-modules/terraform-google-vm/pull/219

Can we open an issue for the `disk-snapshot-local` and null lookup issue so that we can track a fix rather than fixing the example?

**Comment by thiagonache** [comment]
Opened issue https://github.com/terraform-google-modules/terraform-google-vm/issues/220.

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**